saphir/src/file/mod.rs

```diff
@@ -79,19 +80,22 @@ impl File {
 
 impl AsyncRead for File {
     fn poll_read(mut self: Pin<&mut Self>, cx: &mut Context<'_>, buf: &mut [u8]) -> Poll<io::Result<usize>> {
-        self.inner.as_mut().poll_read(cx, buf)
+        let buf_len = buf.len();
+        match self.inner.as_mut().poll_read(cx, &mut ReadBuf::new(buf)) {
+            Poll::Ready(r) => Poll::Ready(r.map(|_| buf.len() - buf_len)),
+            Poll::Pending => Poll::Pending,
+        }
```

Comment:
FileMiddleware always return zero bytes with this.
Fixed with this:
```suggestion
        let mut read_buf = ReadBuf::new(buf);
        let remaining_before = read_buf.remaining();
        match self.inner.as_mut().poll_read(cx, &mut read_buf) {
            Poll::Ready(r) => Poll::Ready(r.map(|_| remaining_before - read_buf.remaining())),
            Poll::Pending => Poll::Pending,
        }
```